modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/BulkActionsDeprecated.js

```diff
@@ -0,0 +1,250 @@
+import ClayLink from '@clayui/link';
+import classNames from 'classnames';
+import {postForm, sub} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useContext, useEffect, useState} from 'react';
+
+import FrontendDataSetContext from '../../FrontendDataSetContext';
+import {OPEN_SIDE_PANEL} from '../../utils/eventsDefinitions';
+import {getOpenedSidePanel} from '../../utils/sidePanels';
+
+function getQueryString(key, values = []) {
+	return `?${key}=${values.join(',')}`;
+}
+
+function getRichPayload(payload, key, values = []) {
+	const richPayload = {
+		...payload,
+		url: payload.baseURL + getQueryString(key, values),
+	};
+
+	return richPayload;
+}
+
+function BulkActions({
+	bulkActions,
+	fluid,
+	items,
+	selectItems,
+	selectedItems,
+	selectedItemsKey,
+	selectedItemsValue,
+	total,
+}) {
+	const {
+		actionParameterName,
+		onBulkActionItemClick,
+		showBulkActionsManagementBar,
+		showBulkActionsManagementBarActions,
+	} = useContext(FrontendDataSetContext);
+
+	const [currentSidePanelActionPayload, setCurrentSidePanelActionPayload] =
+		useState(null);
+
+	function handleActionClick(
+		actionDefinition,
+		formId,
+		formName,
+		loadData,
+		namespace,
+		sidePanelId
+	) {
+		const {data, href, slug, target} = actionDefinition;
+
+		if (target === 'sidePanel') {
+			const sidePanelActionPayload = {
+				baseURL: href,
+				id: sidePanelId,
+				onAfterSubmit: () => loadData(),
+				slug: slug ?? null,
+			};
+
+			Liferay.fire(
+				OPEN_SIDE_PANEL,
+				getRichPayload(
+					sidePanelActionPayload,
+					selectedItemsKey,
+					selectedItemsValue
+				)
+			);
+
+			setCurrentSidePanelActionPayload(sidePanelActionPayload);
+		}
+		else if (onBulkActionItemClick) {
+			onBulkActionItemClick({
+				action: actionDefinition,
+				formId,
+				formName,
+				loadData,
+				namespace,
+				selectedData: {
+					items: selectedItems,
+					keyValues: selectedItemsValue,
+				},
+			});
+		}
+		else if (formId || (formName && namespace)) {
+			const namespacedId = formId || `${namespace}${formName}`;
+
+			const form = document.getElementById(namespacedId);
+
+			if (form) {
+				postForm(form, {
+					data: {
+						...data,
+						[`${actionParameterName || selectedItemsKey}`]:
+							selectedItemsValue.join(','),
+					},
+					url: href || form.action,
+				});
+			}
+		}
+	}
+
+	useEffect(
+		() => {
+			if (!currentSidePanelActionPayload) {
+				return;
+			}
+
+			const currentOpenedSidePanel = getOpenedSidePanel();
+
+			if (
+				currentOpenedSidePanel?.id ===
+					currentSidePanelActionPayload.id &&
+				currentOpenedSidePanel.url.indexOf(
+					currentSidePanelActionPayload.baseURL
+				) > -1
+			) {
+				Liferay.fire(
+					OPEN_SIDE_PANEL,
+					getRichPayload(
+						currentSidePanelActionPayload,
+						selectedItemsValue
+					)
+				);
+			}
+		},
+
+		// eslint-disable-next-line react-compiler/react-compiler
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[selectedItemsValue]
```

Comment:
Uh, this thing may give nightmares to @bryceosterhaus . We will soon delete this file, as we remove FF, right? If that's the case, it's probably ok not to waste time on a possibly difficult refactor to clean this up.

Reply:
Yup, this also isn't new code. It's been in there so it's only getting checked I assume because of this new created file.